nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/adapter/AdminConsoleAdapter.java

```diff
@@ -162,165 +164,161 @@ public HttpHandler getHttpService() {
+        resourceBundle = getResourceBundle(request.getLocale());
 
-        Method method = req.getMethod();
+        Method method = request.getMethod();
         if (!checkHttpMethodAllowed(method)) {
-            res.setStatus(java.net.HttpURLConnection.HTTP_BAD_METHOD,
-                    method.getMethodString() + " " + bundle.getString("http.bad.method"));
-            res.setHeader("Allow", getAllowedHttpMethodsAsString());
+            response.setStatus(HttpStatus.METHOD_NOT_ALLOWED_405);
+            response.setHeader(Header.Allow, getAllowedHttpMethodsAsString());
             return;
         }
-        if (!env.isDas()) {
-            sendStatusNotDAS(req, res);
+
+        if (!serverEnvironment.isDas()) {
+            sendStatusNotAvailable(response, "statusNotDAS.html");
             return;
         }
 
-        //This is needed to support the case where user update to 3.1 from previous release, and didn't run the upgrade tool.
+        if (loadingOption == ConsoleLoadingOption.NEVER) {
+            sendStatusNotAvailable(response, "statusDisabled.html");
+            return;
+        }
+
+        //This is needed to support the case where user update from release prior to 3.1.
         if (adminConsoleConfigUpgrade == null) {
-            adminConsoleConfigUpgrade = habitat.getService(AdminConsoleConfigUpgrade.class);
+            adminConsoleConfigUpgrade = serviceLocator.getService(AdminConsoleConfigUpgrade.class);
         }
 
         try {
-            if (!latch.await(100L, TimeUnit.SECONDS)) {
-                // todo : better error reporting.
+            if (!serverReady.await(100L, TimeUnit.SECONDS)) {
                 logger.log(Level.SEVERE, KernelLoggerInfo.consoleRequestTimeout);
                 return;
             }
-        } catch (InterruptedException ex) {
+        } catch (InterruptedException e) {
             logger.log(Level.SEVERE, KernelLoggerInfo.consoleCannotProcess);
             return;
         }
-        logRequest(req);
-        if (isResourceRequest(req)) {
+
+        logRequest(request);
+
+        if (isResourceRequest(request)) {
             try {
-                handleResourceRequest(req, res);
-            } catch (IOException ioe) {
-                if (logger.isLoggable(Level.SEVERE)) {
-                    logger.log(Level.SEVERE, KernelLoggerInfo.consoleResourceError,
-                            new Object[]{req.getRequestURI(), ioe.toString()});
-                }
-                if (logger.isLoggable(Level.FINE)) {
-                    logger.log(Level.FINE,
-                            ioe.toString(),
-                            ioe);
-                }
+                handleResourceRequest(request, response);
+            } catch (IOException e) {
+                logger.log(Level.SEVERE, KernelLoggerInfo.consoleResourceError, new Object[] {request.getRequestURI(), e});
+                logger.log(Level.FINE, e, e::toString);
```

Comment:
Hmm, I forgot this even exists: `public void log(Level level, Throwable thrown, Supplier<String> msgSupplier) {`